File: BoxComparer.cs

/*****************************************************************************
 *                                                                           *
 *  BoxComparer.cs                                                           *
 *  19 March 2007                                                            *
 *  Project: Crystalline                                                     *
 *  Written by: Richard Sartor                                               *
 *  Copyright © 2007 Metaphysics Industries, Inc.                            *
 *                                                                           *
 *  Converted from C++ to C# on 10 November 2007                             *
 *                                                                           *
 *  A class that sorts boxes within a framework by position.                 *
 *                                                                           *
 *****************************************************************************/

using System;
using System.Collections.Generic;
using MetaphysicsIndustries.Collections;

namespace MetaphysicsIndustries.Crystalline
{
	public class BoxComparer : IComparer<Box>
	{
		public BoxComparer(bool _LeftTopVsRightBottom, bool _Vertical)
		{
			_leftTopVsRightBottom = _LeftTopVsRightBottom;
			_vertical = _Vertical;
		}

		public virtual int Compare(Box x, Box y)
		{
			if (_leftTopVsRightBottom)
			{
				if (_vertical)
				{
					
					return x.Location.Y.CompareTo(y.Location.Y);
				}
				else
				{
					
					return x.Location.X.CompareTo(y.Location.X);
				}
			}
			else
			{
				if (_vertical)
				{
					
					return x.Rect.Bottom.CompareTo(y.Rect.Bottom);
				}
				else
				{
					
					return x.Rect.Right.CompareTo(y.Rect.Right);
				}
			}
		}

        private bool _vertical;
        private bool _leftTopVsRightBottom;
	}
}

File: Crystalline/BoxComparer.cpp

/*****************************************************************************
 *                                                                           *
 *  BoxComparer.cpp                                                          *
 *  19 March 2007                                                            *
 *  Project: Crystalline                                                     *
 *  Written by: Richard Sartor                                               *
 *  Copyright © 2007 Metaphysics Industries                                  *
 *                                                                           *
 *  A class that sorts boxes within a framework by position.                 *
 *                                                                           *
 *****************************************************************************/



#include "stdafx.h"
#include "BoxComparer.h"



namespace MetaphysicsIndustries { namespace Crystalline { 
;



BoxComparer::BoxComparer(bool _LeftTopVsRightBottom, bool _Vertical)
{
	this->lefttopvsrightbottom = _LeftTopVsRightBottom;
	this->vertical = _Vertical;
}

int BoxComparer::Compare(IBox^ x, IBox^ y)
{

	if (this->lefttopvsrightbottom)
	{
		if (this->vertical)
		{
			//top
			return x->Location.Y.CompareTo(y->Location.Y);
		}
		else
		{
			//left
			return x->Location.X.CompareTo(y->Location.X);
		}
	}
	else
	{
		if (this->vertical)
		{
			//bottom
			return x->Rect.Bottom.CompareTo(y->Rect.Bottom);
		}
		else
		{
			//right
			return x->Rect.Right.CompareTo(y->Rect.Right);
		}
	}

	return 0;
}



} } 




